common/src/main/java/com/wynntils/features/user/LowHealthVignetteFeature.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.Window;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.managers.Model;
+import com.wynntils.gui.render.RenderUtils;
+import com.wynntils.gui.render.Texture;
+import com.wynntils.mc.event.ClientTickEvent;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.MathUtils;
+import com.wynntils.wynn.model.ActionBarModel;
+import java.util.List;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class LowHealthVignetteFeature extends UserFeature {
+    private static final float INTENSITY = 0.3f;
+
+    @Config
+    public int lowHealthPercentage = 25;
+
+    @Config
+    public float animationSpeed = 0.6f;
+
+    @Config
+    public HealthVignetteEffect healthVignetteEffect = HealthVignetteEffect.Pulse;
+
+    private float animation = 10f;
+    private float value = INTENSITY;
+
+    @Override
+    public List<Class<? extends Model>> getModelDependencies() {
+        return List.of(ActionBarModel.class);
+    }
+
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    public void onRenderGui(RenderEvent.Post event) {
+        if (event.getType() != RenderEvent.ElementType.GUI) return;
+
+        float healthPercent = (float) ActionBarModel.getCurrentHealth() / ActionBarModel.getMaxHealth();
```

Comment:
You're doing basically the same calculation in onTick. Can't you set a `shouldRender` to `true` there, or something like that?